pkg/bindings/system/system.go

```diff
@@ -42,6 +43,27 @@ func Events(ctx context.Context, eventChan chan types.Event, cancelChan chan boo
 		}()
 	}
 
+	if response.StatusCode != http.StatusOK {
+		var bodyText string
+		if body, err := io.ReadAll(response.Body); err == nil {
+			bodyText = string(body)
+		}
+		var errorModel errorhandling.ErrorModel
+		if err := json.Unmarshal([]byte(bodyText), &errorModel); err == nil && errorModel.Message != "" {
+			bodyText = errorModel.Message
+		}
+		if response.Status != "" {
+			if bodyText != "" {
+				return fmt.Errorf("%s: %s", response.Status, bodyText)
+			}
+			return errors.New(response.Status)
+		}
+		if bodyText != "" {
+			return fmt.Errorf("server responded with unexpected status code %d: %s", response.StatusCode, bodyText)
+		}
```

Comment:
the rest of the code simply calls `return response.Process(nil)` to convert the error

Reply:
Oh, that's _much_ better.  Thanks!